### PR TITLE
[CI] Add check for typos

### PR DESCRIPTION
**Pull request by Flakebi**
Add a github CI workflow that checks for typos.

Example for a failing run: https://github.com/Flakebi/llpc/pull/5

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit ae18682f7a8fc0ae3ad856dafa3fed0388cf2aeb
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 6cc0a035259663610287d8d99262505e78e01db2</li>
<li>SPVGEN: dev efa3cec20c6ebc66254ff9cca9b3ea124811d7a3</li>
<li>XGL: dev 089c9244e6baa5ed42b1342b391c7c6aec521a2f</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev ad9f69a6cf94a03817d1d40bacf1920f31e3c727</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225388)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31425/56347 (55.8%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24922/56347 (44.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31413/56347 (55.7%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24934/56347 (44.3%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31748/56347 (56.3%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24599/56347 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34582/56347 (61.4%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 21765/56347 (38.6%)</li>
</ul></details>


</ul></details>

**Review by kuhar — COMMENTED** [review]
Is there a way to run this locally, outside of CI?

**Comment by Flakebi** [comment]
Yes, just install [typos](https://github.com/crate-ci/typos) and run `typos` in the llpc repo.

**Comment by kuhar** [comment]
> Yes, just install [typos](https://github.com/crate-ci/typos) and run `typos` in the llpc repo.

Can we make the workflow print such instructions when the check step fails?

**Comment by Flakebi** [comment]
I don’t see an easy way to do this. I tried printing instructions in another step after the spell checking, but you have to expand that step explicitly to see the instructions (see https://github.com/Flakebi/llpc/runs/4191321244?check_suite_focus=true, my step name of `Checkout LLPC` is not the best but I think it’s well hidden, even if the name would be better).

**Comment by kuhar** [comment]
> I don’t see an easy way to do this. I tried printing instructions in another step after the spell checking, but you have to expand that step explicitly to see the instructions (see https://github.com/Flakebi/llpc/runs/4191321244?check_suite_focus=true, my step name of `Checkout LLPC` is not the best but I think it’s well hidden, even if the name would be better).

How about we print those instructions before we run the check step? I'd just like to have something that once can refer to if they see this check fail on their PR.

File: .github/workflows/check-typos.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      uses: actions/checkout@v2
+
+    - name: Check spelling
+      uses: crate-ci/typos@v1.2.1
+
+    - name: Help
+      if: always()
+      run: |
+        echo 'To run the check locally, install typos from https://github.com/crate-ci/typos and run `typos` in the repository.'
+        echo 'Run `typos -w` to automatically correct spelling mistakes.'
```
Comment (kuhar) [inline .github/workflows/check-typos.yml:19]: Don't we have to escape those ticks like in https://github.com/GPUOpen-Drivers/llpc/pull/1495/files?

**Comment by Flakebi** [comment]
I named the step “Help”, hopefully that makes it attractive for people to open it.

File: .github/workflows/check-typos.yml (hunk 1 above)
Comment (Flakebi) [reply .github/workflows/check-typos.yml:19]: No, it’s single quotes here, so it works: https://github.com/GPUOpen-Drivers/llpc/runs/4191968589?check_suite_focus=true

**Review by kuhar — APPROVED** [review]
LGTM. I'm dyslexic and make tons of typos, tools like this are very helpful.

**Review by amdrexu — APPROVED** [review]

**Pull request merged**